### PR TITLE
Fix video_input_image_list config reflection

### PR DESCRIPTION
**Pull request by Purg**
Fixed config getter to use currently set values instead of hard-coded
defaults. This previously would wipe out custom values when performing
iterative configuration reflection.

### CODE REVIEW

**Review by mwoehlke-kitware — COMMENTED** [review]
Although it would be "more work", I wonder if it wouldn't be more pedantically correct to reconstitute the values from the parsed lists?

**Comment by Purg** [comment]
> Although it would be "more work", I wonder if it wouldn't be more pedantically correct to reconstitute the values from the parsed lists?

I did consider that, but thought I would start with the simpler solution as a first pass. I don't want to over-engineer this.

I am not sure what you refer to as "pedantically correct", so I should clarify what I would have expected config-input/config-getting behavior to be. I expected that if I configure a `video_input_image_list` with some "path", at any point when I get a config back out I should get the same "path." The `c_search_path` vector is modified at runtime, but I'm not sure I consider those modifications to be "persistent" in that they should be exposed back into a configuration file: the modifications only seem to be needed locally at runtime and likely don't need to actually be retained in state (`"."` is constant, and `open` file parent directories are only relevant during the current invocation of `read_from_file`).

From my perspective, retaining the raw as-input string *is* "pedantic". I would justify the additional string storage with their content is not expected to be very long and `video_input_image_list` algorithm instances are not expected to be copied greatly or be instantiated in any great quantity.

**Comment by mwoehlke-kitware** [comment]
For example, if the user passes "a:b:c::c:b::d", the algorithm will use {"a", "b", "c", "d"} (well, I added de-duplication also, but if the algorithm were to start doing that, it would make this point even more relevant). Effectively, then, the algorithm is using "a:b:c:d". Note particularly the lack of empty entries.

The search path is a trickier case because we probably should *not* export e.g. the appended ".". In any case, I'm not worried about the extra storage; the thought was more if we should export the "sanitized" version that is a closer representation to what is actually used.

Is it *important*? Maybe not... was just a thought...

**Comment by Purg** [comment]
I found some bandwidth in the middle of a meeting that I was just listening in on. I have updated the implementation to reconstruct appropriately from stored vectors.

File: arrows/core/video_input_image_list.cxx [hunk 1]
```diff
@@ -145,13 +163,15 @@ ::set_configuration( vital::config_block_sptr in_config )
 
   // Extract string and create vector of directories
   auto const& path = config->get_value< std::string >( "path", {} );
-  kv::tokenize( path, d->c_search_path, ":", kv::TokenizeTrimEmpty );
+  d->c_search_path.clear();  // make sure vec is empty before populating it
+  kv::tokenize( path, d->c_search_path, SEP_PATH, kv::TokenizeTrimEmpty );
```
Comment (mwoehlke-kitware) [inline arrows/core/video_input_image_list.cxx:167]: OT since it isn't a change, but it seems like this is broken on Windows? 🙂

**Review by mwoehlke-kitware — APPROVED** [review]
LGTM, thanks!

**Pull request merged**